### PR TITLE
Feature swagger: name mangling in docs

### PR DESCRIPTION
**Pull request by lukaszkorecki**
Fixes the Swagger generator, and ensures that if name mangling (aka "use kebab case for keys/enums") is ON, then the generated schema follows that.

- [x] When `mangle-names/kebab-case-keys` is set to  `true` - generated docs also have to reflect that
- [x] **Fixes a major bug**: we were automatically serializing Duckula response to JSON, which prevents it from being usable with Ring middlewares which process output (for example to augment the response body post validation). This has impact on our internal applications as we will have to ensure we include 'json-response-body' middleware, to ensure we automatically serialize response body to JSON. 
- [x] Fix reflection warnings in the Swagger generator

While not yet there, this also paves the way for automatically generating Swagger docs from Duckula config, sidestepping the Prismatic Schema conversion step. Not quite there yet, but we'll get there!

### CODE REVIEW

File: src/duckula/middleware.clj [hunk 1]
```diff
@@ -12,7 +12,8 @@
   [handler-fn]
   (-> handler-fn
       (ring-json/wrap-json-body {:keywords? true})
-      (ring-defaults/wrap-defaults ring-defaults/api-defaults)))
+      (ring-defaults/wrap-defaults ring-defaults/api-defaults)
+      (ring-json/wrap-json-response)))
```
Comment (lukaszkorecki) [inline src/duckula/middleware.clj:16]: This is a breaking change! Our internal middleware stack will have to make similar change - otherwise we will get Ring errors. 
That said, this makes the whole setup much cleaner and makes Duckula behave more like a Ring middleware because we do not force JSON as output format.

File: src/duckula/avro/schema.clj [hunk 2]
```diff
@@ -34,27 +37,37 @@
     ;; Complex Types
     Schema$Type/RECORD
     (apply hash-map (mapcat (fn [key val]
-                              [(keyword key) val])
-                            (map #(.name %) (.getFields avro-schema))
-                            (map #(->map (.schema %)) (.getFields avro-schema))))
+                              (let [k (if mangle-names?
+                                        (-> key (str/replace \_ \-) keyword)
+                                        (keyword key))]
+                                [k val]))
```
Comment (AfonsoTsukamoto) [inline src/duckula/avro/schema.clj:43]: no need for the let:

```clj
[(if mangle-names?
    (-> key (str/replace \_ \-) keyword)
    (keyword key))
  val]
```

That said, this function is still a bit hard to follow

File: src/duckula/avro/schema.clj [hunk 3]
```diff
@@ -34,27 +37,37 @@
...
+                            (map #(->map {:avro-schema (.schema ^Schema$Field %)
+                                          :mangle-names? mangle-names?})
+                                 (.getFields ^Schema avro-schema))))
 
     Schema$Type/ENUM
-    (apply s/enum (.getEnumSymbols avro-schema))
+    (apply s/enum (map (fn [enum]
+                         (if mangle-names?
+                           (str/replace enum \_ \-)
+                           enum))
```
Comment (AfonsoTsukamoto) [inline src/duckula/avro/schema.clj:53]: Since is also used above, should this be a function?

**Review by AfonsoTsukamoto — APPROVED** [review]

File: src/duckula/avro/schema.clj (hunk 2 above)
Comment (lukaszkorecki) [reply src/duckula/avro/schema.clj:43]: I can extract bits of it, but it will largely be unchanged - it's recursive in nature, so not much I can do about, even after we skip the step of generating Prismatic Schema.

Regarding the suggestion - that was the code I had originally, I found it harder to read because of the `if` being inside of a vector.

File: src/duckula/avro/schema.clj (hunk 3 above)
Comment (lukaszkorecki) [reply src/duckula/avro/schema.clj:53]: Definitely!
